### PR TITLE
State Sync & t*-db

### PR DESCRIPTION
**Pull request by faddat**
This PR adds a state sync script for gaia v8, and avoids the t* github org, while adding support for pebble-db.

to experience velocity:

```bash
bash contrib/statesync.bash
```

### CODE REVIEW

**Comment by mpoke** [comment]
@faddat Thanks for the contribution. Could you please change the base branch to main?

**Comment by faddat** [comment]
Oh yeah absolutely

**Comment by faddat** [comment]
Sorry about the delay here

Also it's possible that we might want to just change over to comet DB but for comet DB I would really love to see the pebble changes merged.

File: contrib/statesync.bash [hunk 1]
```diff
@@ -1,33 +1,48 @@
...
+## USAGE RUNDOWN
+# Not for use on live nodes
+# For use when testing.
+# Assumes that ~/.evmosd doesn't exist
+# can be modified to suit your purposes if ~/.evmosd does already exist
+
 set -uxe
 
 # Set Golang environment variables.
 export GOPATH=~/go
```
Comment (mmulji-ic) [inline contrib/statesync.bash:15]: I would check whether the GOPATH is already set. If not advise the user to set it and add it to that path. 
I use Homebrew and that installs golang to a totally different bin directory.

File: contrib/statesync.bash [hunk 2]
```diff
@@ -1,33 +1,48 @@
...
 export PATH=$PATH:~/go/bin
 
-# Install Gaia.
-make install
+# Install with pebbledb (uncomment for incredible performance)
+# go mod edit -replace github.com/tendermint/tm-db=github.com/baabeetaa/tm-db@pebble
+# go mod tidy
+go install -ldflags '-w -s -X github.com/cosmos/cosmos-sdk/types.DBBackend=pebbledb -X github.com/tendermint/tm-db.ForceSync=1' -tags pebbledb ./...
```
Comment (mmulji-ic) [inline contrib/statesync.bash:21]: If you want users to have the easiest experience, why install pebbledb and not use the default.
I would leave this commented out.
SDK are also phasing out some databases ...  do we want to include all of these, especially for new users who just want to try out Gaia and sync easily?

File: contrib/statesync.bash [hunk 3]
```diff
@@ -36,12 +51,13 @@ echo "trust_hash: $TRUST_HASH"
...
 export GAIAD_STATESYNC_TRUST_HASH=$TRUST_HASH
 
 # Fetch and set list of seeds from chain registry.
-export GAIAD_P2P_SEEDS=$(curl -s https://raw.githubusercontent.com/cosmos/chain-registry/master/cosmoshub/chain.json | jq -r '[foreach .peers.seeds[] as $item (""; "\($item.id)@\($item.address)")] | join(",")')
+GAIAD_P2P_SEEDS=$(curl -s https://raw.githubusercontent.com/cosmos/chain-registry/master/cosmoshub/chain.json | jq -r '[foreach .peers.seeds[] as $item (""; "\($item.id)@\($item.address)")] | join(",")')
+export GAIAD_P2P_SEEDS
 
 # Start chain.
-gaiad start --x-crisis-skip-assert-invariants
+gaiad start --x-crisis-skip-assert-invariants --db_backend pebbledb --iavl-disable-fastnode false
```
Comment (mmulji-ic) [inline contrib/statesync.bash:63]: May need changing depending on db selection

File: contrib/statesync.bash [hunk 4]
```diff
@@ -36,12 +51,13 @@ echo "trust_hash: $TRUST_HASH"
...
 export GAIAD_STATESYNC_ENABLE=true
 export GAIAD_P2P_MAX_NUM_OUTBOUND_PEERS=200
-export GAIAD_STATESYNC_RPC_SERVERS="https://rpc.cosmos.network:443,https://rpc.cosmos.network:443"
+export GAIAD_STATESYNC_RPC_SERVERS="https://cosmos-rpc.polkachu.com:443,https://rpc-cosmoshub-ia.cosmosia.notional.ventures:443"
 export GAIAD_STATESYNC_TRUST_HEIGHT=$BLOCK_HEIGHT
 export GAIAD_STATESYNC_TRUST_HASH=$TRUST_HASH
 
 # Fetch and set list of seeds from chain registry.
-export GAIAD_P2P_SEEDS=$(curl -s https://raw.githubusercontent.com/cosmos/chain-registry/master/cosmoshub/chain.json | jq -r '[foreach .peers.seeds[] as $item (""; "\($item.id)@\($item.address)")] | join(",")')
+GAIAD_P2P_SEEDS=$(curl -s https://raw.githubusercontent.com/cosmos/chain-registry/master/cosmoshub/chain.json | jq -r '[foreach .peers.seeds[] as $item (""; "\($item.id)@\($item.address)")] | join(",")')
```
Comment (mmulji-ic) [inline contrib/statesync.bash:59]: This is really helpful!

File: go.mod [hunk 5]
```diff
@@ -305,9 +319,11 @@ replace (
...
 	// Use Informal Systems fork of Tendermint
-	github.com/tendermint/tendermint => github.com/informalsystems/tendermint v0.34.25
+	github.com/tendermint/tendermint => github.com/informalsystems/tendermint v0.34.26
 
 	// latest grpc doesn't work with with our modified proto compiler, so we need to enforce
 	// the following version across all dependencies.
 	google.golang.org/grpc => google.golang.org/grpc v1.33.2
 )
+
+replace github.com/tendermint/tm-db => github.com/notional-labs/tm-db v0.6.8-0.20230214123107-77c191d0f7ee
```
Comment (mmulji-ic) [inline go.mod:329]: Is there not a normal commit version for tm-db?

File: go.mod [hunk 6]
```diff
@@ -293,7 +307,7 @@ replace (
 	github.com/confio/ics23/go => github.com/confio/ics23/go v0.9.0
 
 	// enforce same SDK and IBC on all dependencies
-	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.45.12
+	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.45.13
```
Comment (mmulji-ic) [inline go.mod:310]: Question/Reminder, Do we want to keep this script in sync with main/whichever release branch we are using?

File: contrib/statesync.bash (hunk 2 above)
Comment (faddat) [reply contrib/statesync.bash:21]: sure

**Comment by faddat** [comment]
So, since we are on v9 now, I don't think it makes too much sense to update the sync script for v8 in fact.

**Pull request closed without merging**